modules/core/src/v2/wallet.ts

```diff
@@ -1341,19 +1359,23 @@ export class Wallet {
         }
 
         newAddress.keychains = keychains;
-        newAddress.baseAddress = _.get(self._wallet, 'coinSpecific.baseAddress');
+        newAddress.baseAddress = !_.isUndefined(baseAddress) ? baseAddress : _.get(self._wallet, 'coinSpecific.baseAddress');
 
         const verificationData: VerifyAddressOptions = _.merge({}, newAddress, { rootAddress });
 
         if (verificationData.error) {
           throw new AddressGenerationError(verificationData.error);
         }
 
+        const impliedForwarderVersion = !_.isUndefined(addressParams.forwarderVersion) ? addressParams.forwarderVersion :
+        verificationData.coinSpecific?.forwarderVersion;
```

Comment:
please add
```
/**
 * @prettier
 */
```
 at the top and format this file

Reply:
Sure Otto, will add this once the file is ready to merge. Adding prettier does a lot of reformatting to the code which will make it really hard to review.

Reply:
not a problem because we can review individually

Reply:
Done